### ISSUE TITLE
data race in TestErrConnKilled

### ISSUE THREAD

**Issue by pacoxu** [kubernetes/kubernetes#103246]
#### Which jobs are flaking:
pull-kubernetes-unit

#### Which test(s) are flaking:
TestErrConnKilled

#### Testgrid link:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103107/pull-kubernetes-unit/1409355697006579712

#### Reason for failure:
Data RACE in unit test
```
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters: TestErrConnKilled expand_less	0s
=== RUN   TestErrConnKilled
==================
WARNING: DATA RACE
Write at 0x0000039a5818 by goroutine 405:
  k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.captureStdErr.func2()      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout_test.go:220 +0x5a
  k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.TestErrConnKilled()     /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout_test.go:270 +0x411
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1193 +0x202

Previous read at 0x0000039a5818 by goroutine 223:
  k8s.io/kubernetes/vendor/k8s.io/klog/v2.(*loggingT).output()
      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:924 +0xb74
  k8s.io/kubernetes/vendor/k8s.io/klog/v2.(*loggingT).printDepth()
      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:735 +0x2a4
  k8s.io/kubernetes/vendor/k8s.io/klog/v2.ErrorDepth()
      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/klog/v2/klog.go:1456 +0x12d
  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.logError()      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:114 +0x2f
  k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleError()      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:108 +0xa1
  k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func2.1()      /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:135 +0x273
```

#### Anything else we need to know:
- links to go.k8s.io/triage appreciated
- links to specific failures in spyglass appreciated

https://storage.googleapis.com/k8s-gubernator/triage/index.html?test=TestErrConnKilled#aad8ed4d8191a2a68229

**Comment by fedebongio** [comment]
Thank you @learner0810 for taking a look, let us know when you have a PR!
/triage accepted

**Comment by pacoxu** [comment]
It dups with #101097 and should be fixed by #101098.

/close
